newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -93,97 +91,103 @@ export default class SemiControlledAutoComplete extends React.Component<
       <MarkdownText source={this.props.helperMarkdownText} />
     ) : null;
 
-    const handleOptionLabel = (option: DataSource | null): string => {
-      if (option.value) {
-        return option.value;
-      } else {
-        return currentInputValue();
-      }
+    const getOptionLabel = (option: Option): string => {
+      // option.value ? option.value : getCurrentInputValue();
+      // Writing shorthand will fail flow checks.
+      // Explicitly define all the cases of type Option.
+      if (option.value) return option.value;
+      else return getCurrentInputValue();
     };
 
-    const handleChange = (event: Event, option: DataSource | string): void => {
-      if (option !== null) {
-        if (option.value) {
+    const handleChange = (event, option: Option | string): void => {
+      // Explicitly handle option.type to avoid flow errors
+      if (option !== null && option.type !== 'separator') {
+        if (typeof option === 'string') {
+          this.props.onChange(option);
+        } else {
           if (option.onClick) option.onClick();
           else {
             if (this.props.onChoose) this.props.onChoose(option.value);
             else this.props.onChange(option.value);
           }
-        } else {
-          /* option is just a string in this case, and it
-             enables user to type something other than provided options. */
-
-          this.props.onChange(option);
         }
-        if (event.key) handleKeyDown(event.key);
-        if (this.props.onRequestClose) this.props.onRequestClose();
       }
+      if (typeof event.key === 'string') handleKeyDown(event.key);
+      if (this.props.onRequestClose) this.props.onRequestClose();
     };
 
-    //Todo: Find a way to call this function everytime.
     const handleKeyDown = (key: string): void => {
       if (key === 'Enter' && this._input.current !== null)
         this._input.current.blur();
     };
 
-    const handleInputChange = (
-      e: Event,
-      value: string,
-      reason: string
-    ): void => {
+    const handleInputChange = (e, value: string, reason: string): void =>
       this.setState({ inputValue: value });
-    };
 
-    const nullifyDefaultStyling = (params: Object): void => {
-      const { InputProps, inputProps, InputLabelProps } = params;
+    const getDefaultStylingProps = (params: Object): Object => {
+      var { InputProps, inputProps, InputLabelProps, ...other } = params;
       InputProps.className = null;
       InputProps.endAdornment = null;
       inputProps.className = null;
-      InputLabelProps.disableAnimation = true;
       InputProps.style = styles.inputRoot;
       inputProps.style = styles.inputInput;
+      inputProps = {
+        ...inputProps,
+        onKeyDown: (e): void => {
+          if (e.key === 'Escape' && this.props.onRequestClose)
+            this.props.onRequestClose();
+        },
+      };
+      return { InputProps, inputProps, InputLabelProps, ...other };
```

Comment:
Prefer a immutable way of doing things:

return {
  InputProps: {
    ...InputProps,
    className: null,
    endAdornment: null,
    style: styles.inputRoot
  },
  inputProps: {
    ...inputProps
    onKeyDown: (event) => // etc..
  },
  InputLabelProps
  // etc...
}